cpp/include/cudf/io/detail/json.hpp

```diff
@@ -69,11 +69,20 @@ void normalize_single_quotes(datasource::owning_buffer<rmm::device_buffer>& inda
  * @brief Normalize unquoted whitespace (space and tab characters) using FST
  *
  * @param indata Input device buffer
+ * @param col_lengths Length of contents of each row in column
+ * @param col_offsets Offsets to column contents in input buffer
```

Comment:
Why do we need 2 separate arrays for lengths and offsets? Can lengths be computed from offsets as `offsets[i+1] - offset[i]`? Doing so we can reduce memory usage.

Reply:
in this case, it does not. the offsets are spread throughout the json string. This offset is not coming from a string column.

Reply:
echoing Karthikeyan's response above - we need to first gather the required strings from `d_input` using the lengths and offsets information for the FST. The `cub::DeviceCopy::Batched` performs this operation.